src/elements/content-sidebar/hooks/useSidebarMetadataFetcher.ts

```diff
@@ -174,6 +190,41 @@ function useSidebarMetadataFetcher(
         [api, file, onApiError],
     );
 
+    const extractSuggestions = React.useCallback(
+        async (templateKey: string, fields: MetadataTemplateField[]) => {
+            const aiAPI = api.getIntelligenceAPI();
+
+            await wait(1000);
```

Comment:
Why do we have a setTimeout here? We should have a constant for the delay time as well.